ttnn/cpp/ttnn/operations/eltwise/binary/device/binary_composite_op.hpp

```diff
@@ -117,20 +150,76 @@ struct OpHandler_IsClose<BinaryCompositeOpType::ISCLOSE> {
     }
 };
 
+
+template <>
+struct OpHandler_Div<BinaryCompositeOpType::DIV> {
+    static Tensor handle(const Tensor& t1, const Tensor& t2, bool accurate_mode, string round_mode, const std::optional<MemoryConfig>& mem_cfg) {
+        return _div(t1, t2, accurate_mode, round_mode, mem_cfg);
+    }
+    static Tensor handle(const Tensor& t1, float value, bool accurate_mode, string round_mode, const std::optional<MemoryConfig>& mem_cfg) {
+        return _div_overload(t1, value, accurate_mode, round_mode, mem_cfg);
+    }
+};
+
+
+template <>
+struct OpHandler_Overload<BinaryCompositeOpType::DIV_NO_NAN> {
+    static Tensor handle(const Tensor& t1, const Tensor& t2, const std::optional<MemoryConfig>& mem_cfg) {
+        return _div_no_nan(t1, t2, mem_cfg);
+    }
+    static Tensor handle(const Tensor& t1, float value, const std::optional<MemoryConfig>& mem_cfg) {
+        return _div_no_nan_overload(t1, value, mem_cfg);
+    }
+};
+
+
+template <>
+struct OpHandler_Overload<BinaryCompositeOpType::FLOOR_DIV> {
+    static Tensor handle(const Tensor& t1, const Tensor& t2, const std::optional<MemoryConfig>& mem_cfg) {
+        return _floor_div(t1, t2, mem_cfg);
+    }
+    static Tensor handle(const Tensor& t1, float value, const std::optional<MemoryConfig>& mem_cfg) {
+        return _floor_div_overload(t1, value, mem_cfg);
+    }
+};
+
+
 // Template functions to get the function pointers
 template <BinaryCompositeOpType OpType>
-auto get_function_type0() {
+auto get_function() {
     return &OpHandler<OpType>::handle;
 }
 
 template <BinaryCompositeOpType OpType>
-auto get_function_type1() {
+auto get_function_with_float() {
     return &OpHandler_Float<OpType>::handle;
 }
 
 template <BinaryCompositeOpType OpType>
-auto get_function_type2() {
+auto get_function_isclose() {
     return &OpHandler_IsClose<OpType>::handle;
 }
 
+using HandleFunctionPtr1 = Tensor (*)(const Tensor&, const Tensor&, const std::optional<MemoryConfig>&);
+using HandleFunctionPtr2 = Tensor (*)(const Tensor&, float, const std::optional<MemoryConfig>&);
+template <BinaryCompositeOpType OpType>
+auto get_function_divnonan_floordiv() -> HandleFunctionPtr1 {
+    return &OpHandler_Overload<OpType>::handle;
+}
+template <BinaryCompositeOpType OpType>
+auto get_function_divnonan_floordiv_overload() -> HandleFunctionPtr2 {
```

Comment:
if these 2 fns are used by both divnonan and floordiv ops, make it common like, get_div_ops_fn, get_div_ops_fn_overload